### PR TITLE
test(cli): add check for missing init files

### PR DESCRIPTION
**Pull request by anshbansal**
CLI rc release failed because of missing `__init__` file at https://github.com/acryldata/datahub/actions/runs/4222580108/jobs/7331259576. Adding check so the problem is detected during PRs itself.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: .github/scripts/check_packages.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import pprint
+
+where = "./src"
+package_list = set(setuptools.find_packages(where))
+namespace_packages = set(setuptools.find_namespace_packages(where))
+diff1 = package_list.difference(namespace_packages)
+if len(diff1) > 0:
+    pprint.pprint(diff1)
+diff2 = namespace_packages.difference(package_list)
+if len(diff2) > 0:
```
Comment (hsheth2) [inline .github/scripts/check_packages.py:11]: Let’s just implement this as part of pytest instead of adding another workflow

Comment (hsheth2) [reply .github/scripts/check_packages.py:11]: The added benefit is that pytest will generate the debug diff details on failure automatically

Comment (anshbansal) [reply .github/scripts/check_packages.py]: done

File: metadata-ingestion/tests/unit/test_packages.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import setuptools
+
+def test_package_list_match_inits():
+    where = "./src"
+    package_list = set(setuptools.find_packages(where))
+    namespace_packages = set(setuptools.find_namespace_packages(where))
+    diff1 = package_list.difference(namespace_packages)
```
Comment (hsheth2) [inline metadata-ingestion/tests/unit/test_packages.py:7]: Just assert that the two sets are equal to each other

Also maybe add a comment that diffs are likely due to missing init files

Comment (anshbansal) [reply metadata-ingestion/tests/unit/test_packages.py]: done

**Pull request merged**